django/test/testcases.py

```diff
@@ -266,6 +275,30 @@ def _remove_databases_failures(cls):
                 method = getattr(connection, name)
                 setattr(connection, name, method.wrapped)
 
+    @classmethod
+    def ensure_connection_patch_method(cls):
+        def patched_ensure_connection(self):
+            if (
+                self.connection is None
+                and self.alias not in cls.databases
+                and self.alias != NO_DB_ALIAS
+            ):
+                # Connection has not yet been established, but the alias is not allowed.
+                message = cls._disallowed_database_msg % {
+                    "test": "%s.%s" % (cls.__module__, cls.__qualname__),
+                    "alias": self.alias,
+                    "operation": "threaded connections",
+                }
+                return _DatabaseFailure(self.ensure_connection, message)()
+
+            # Copied from BaseDatabaseWrapper.ensure_connection as the original as
+            # been patched at the base class level.
+            if self.connection is None:
+                with self.wrap_database_errors:
+                    self.connect()
```

Comment:
What about 3rd-party database backends with the custom `ensure_connection()` implementation?

Reply:
Also, the original `ensure_connection()` is `@async_unsafe` decorated.

Reply:
Of course, copying the source code is a bad idea here 😛 
Let's call the actual `BaseDatabaseWrapper.ensure_connection` method.

Now, the monkey-patching approach means that, as long as we call `BaseDatabaseWrapper.ensure_connection`, the behaviour should work as expected.
Meaning that the Sentry implementation should keep working I believe.
But if another backend overrides `ensure_connection` without calling `super().ensure_connection(...)`, then the behaviour remains the same as today: database connections still work in threads.

Would there be another method(s) that we can leverage? 🤔